contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,143 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
 RETURNS sys.NVARCHAR
 AS 'babelfishpg_tsql', 'tsql_json_query' LANGUAGE C IMMUTABLE PARALLEL SAFE;
 
+-- Function to convert the path format into expected jsonb_set path format using Regular Expression
+CREATE OR REPLACE FUNCTION sys.regex_replace(json_path TEXT)
+RETURNS TEXT
+AS
+$BODY$
+BEGIN
+    SELECT regexp_replace(json_path, '\$\.|]|\$\[' , '' , 'ig') INTO json_path; -- To remove the "$." and "]" sign from the string 
+    SELECT regexp_replace(json_path, '\.|\[' , ',' , 'ig') INTO json_path; -- To replace the "." and "[" with the "," to change into required format
+    SELECT CONCAT('{',json_path,'}') INTO json_path; -- Final required format of path by jsonb_set
+    return json_path;
+END
+$BODY$
+LANGUAGE plpgsql;
+
+--Function to set the append_modifier and create_if_missing flags
+CREATE TYPE falgs_set AS (create_if_missing BOOL, append_modifier BOOL);
```

Comment:
typo here; should be `flags_set`, not `falgs_set`

Reply:
Also, the name of the datatype is not very descriptive - what if someone else wants to create a flag type? It should be more specific to its use, like `sys.json_modify_flags`. Also, make sure to declare this in the sys schema.

Reply:
Have made the changes and named the datatype sys.json_modify_flags